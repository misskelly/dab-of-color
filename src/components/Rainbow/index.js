/* eslint-disable camelcase */
import React from 'react';

export const Rainbow = props => {
  const { color_1, color_2, color_3, color_4, color_5 } = props.colors;
  return (
    <li className="rainbow">
      <svg
        xmlns="http://www.w3.org/2000/svg"
        x="0"
        y="0"
        enableBackground="new 0 0 60 60"
        version="1.1"
        viewBox="0 0 60 46"
        xmlSpace="preserve"
      >
        <path
          fill={color_1}
          d="M4.645 45.577a25.986 25.986 0 01-.029-1.154c0-13.997 11.387-25.385 25.385-25.385s25.385 11.387 25.385 25.385c0 .387-.012.771-.029 1.154h4.616c.012-.384.027-.767.027-1.154 0-16.569-13.431-30-30-30s-30 13.431-30 30c0 .387.015.77.029 1.154h4.616z"
        />
        <path
          fill={color_2}
          d="M9.289 45.577c-.021-.384-.058-.764-.058-1.154 0-11.452 9.317-20.769 20.769-20.769s20.769 9.318 20.769 20.769c0 .389-.037.77-.058 1.154h4.645c.017-.383.029-.767.029-1.154 0-13.997-11.387-25.385-25.385-25.385S4.615 30.426 4.615 44.423c0 .387.012.771.029 1.154h4.645z"
        />
        <path
          fill={color_3}
          d="M13.905 45.577c-.027-.383-.058-.764-.058-1.154 0-8.907 7.246-16.154 16.154-16.154s16.154 7.246 16.154 16.154c0 .389-.031.771-.058 1.154h4.615c.021-.384.058-.764.058-1.154 0-11.452-9.317-20.769-20.769-20.769s-20.77 9.317-20.77 20.769c0 .389.037.77.058 1.154h4.616z"
        />
        <path
          fill={color_4}
          d="M18.52 45.577a11.58 11.58 0 01-.058-1.154c0-6.362 5.176-11.538 11.538-11.538s11.538 5.177 11.538 11.538c0 .389-.021.774-.058 1.154h4.615c.027-.383.058-.764.058-1.154 0-8.907-7.246-16.154-16.154-16.154s-16.154 7.246-16.154 16.154c0 .389.031.771.058 1.154h4.617z"
        />
        <path
          fill={color_5}
          d="M23.181 45.577a6.859 6.859 0 01-.104-1.154c0-3.818 3.105-6.923 6.923-6.923s6.923 3.105 6.923 6.923c0 .394-.041.778-.104 1.154h4.661c.038-.38.058-.764.058-1.154 0-6.362-5.176-11.538-11.538-11.538s-11.538 5.177-11.538 11.538c0 .389.021.774.058 1.154h4.661z"
        />
      </svg>
    </li>
  );
};

export default Rainbow;
