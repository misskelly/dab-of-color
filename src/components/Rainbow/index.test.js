import React from 'react';
import { shallow } from 'enzyme';
import Rainbow from '.';

describe('Rainbow', () => {
  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<Rainbow />);

    expect(wrapper).toMatchSnapshot();
  });
});
