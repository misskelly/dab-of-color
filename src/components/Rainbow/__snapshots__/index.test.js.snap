// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rainbow should match the snapshot 1`] = `ShallowWrapper {}`;
